app/controllers/admin/project_steps_controller.rb

```diff
@@ -59,18 +59,24 @@ def update
      # Detect potential schedule shift.
     days_shifted = @step.pending_days_shifted
 
+    # Detect if duration was changed
+    duration_changed = @step.pending_duration_change?
+
     valid = @step.save
 
     # Ignore schedule shift if not successfully saved
     days_shifted = 0 unless valid
 
+    options = {id: @step.id, days_shifted: days_shifted, duration_changed: duration_changed}
+
     if %w(timeline_table calendar).include?(params[:context])
-      valid ? render(json: {id: @step.id, days_shifted: days_shifted}) : render_modal_content(422)
+      valid ? render(json: options) : render_modal_content(422)
     else
       render partial: '/admin/project_steps/project_step', locals: {
         step: @step,
         mode: valid ? :show : :edit,
         days_shifted: days_shifted,
+        duration_changed: duration_changed,
```

Comment:
Is this used anywhere? I'm thinking not. Which is fine I think since this is for the timeline list view. So I don't think this change will work there, but that's ok in my book because I still really want to get rid of this old code!!! But I think you can remove this line as it's not doing anything presently.